src/Equinor.ProCoSys.IPO.Command/PersonCommands/UpdateSavedFilter/UpdateSavedFilterCommandHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.IPO.Domain;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.PersonAggregate;
+using MediatR;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Command.PersonCommands.UpdateSavedFilter
+{
+    public class UpdateSavedFilterCommandHandler : IRequestHandler<UpdateSavedFilterCommand, Result<string>>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly ICurrentUserProvider _currentUserProvider;
+        private readonly IPersonRepository _personRepository;
+
+        public UpdateSavedFilterCommandHandler(
+            IUnitOfWork unitOfWork,
+            ICurrentUserProvider currentUserProvider,
+            IPersonRepository personRepository)
+        {
+            _unitOfWork = unitOfWork;
+            _currentUserProvider = currentUserProvider;
+            _personRepository = personRepository;
+        }
+
+        public async Task<Result<string>> Handle(UpdateSavedFilterCommand request, CancellationToken cancellationToken)
+        {
+            var currentUserOid = _currentUserProvider.GetCurrentUserOid();
+            var person = await _personRepository.GetWithSavedFiltersByOidAsync(currentUserOid);
+            var savedFilter = person.SavedFilters.Single(sf => sf.Id == request.SavedFilterId);
+
+            if (request.DefaultFilter == true)

```

Comment:
will this throw an exception since DefaultFilter is optional?

Reply:
Better: if (request.DefaultFilter.HasValue && request.DefaultFilter.Value)